packages/paima-sdk/paima-events/src/app-events.ts

```diff
@@ -0,0 +1,200 @@
+import type { LogEvent, LogEventFields } from './types.js';
+import type { genEvent } from './types.js';
+import { toPath, TopicPrefix } from './types.js';
+import type { TSchema, Static } from '@sinclair/typebox';
+import { Value } from '@sinclair/typebox/value';
+import { keccak_256 } from 'js-sha3';
+
+export function toSignature<T extends LogEvent<LogEventFields<TSchema>[]>>(event: T): string {
+  return event.name + '(' + event.fields.map(f => f.type.type).join(',') + ')';
+}
+
+export function toTopicHash<T extends LogEvent<LogEventFields<TSchema>[]>>(event: T): string {
+  return keccak_256(toSignature(event));
+}
+
+// this generates the type for the events import.
+export const generateAppEvents = <
```

Comment:
To contextualize this. The contents of  `events/index.ts`  would have to be something like:

```ts
import { genEvent } from '@paima/events';
import { Type } from '@sinclair/typebox';
import { generateAppEvents } from '@paima/events';

const StfEvent1 = genEvent({
  name: 'StfEvent',
  fields: [
    {
      indexed: true,
      name: 'tag',
      type: Type.String(),
    },
    {
      indexed: false,
      name: 'data',
      type: Type.String(),
    },
  ],
} as const);

const StfEvent2 = genEvent({
  name: 'StfEvent',
  fields: [
    {
      indexed: true,
      name: 'tag',
      type: Type.String(),
    },
    {
      indexed: false,
      name: 'data',
      type: Type.String(),
    },
    {
      indexed: false,
      name: 'extraData',
      type: Type.String(),
    },
  ],
});

const StfEvent3 = genEvent({
  name: 'StfEvent',
  fields: [
    {
      indexed: true,
      name: 'tag',
      type: Type.String(),
    },
    {
      indexed: false,
      name: 'data',
      type: Type.String(),
    },
    {
      indexed: false,
      name: 'extraData',
      type: Type.String(),
    },
    {
      indexed: true,
      name: 'block',
      type: Type.Integer(),
    },
  ],
});

const eventDefinitions = [StfEvent1, StfEvent2, StfEvent3] as const;

export const events = generateAppEvents(eventDefinitions);
```

(the contents don't really matter here, it's just part of my current test setup, but it's not finished).